File: sentimental-cash/cash.py
import cs50

while True:
    dollars = cs50.get_float('Change: ')
    if dollars > 0:
        coins = 0

        quarters = int(dollars / 0.25)
        dime = int((dollars % 0.25) / 0.10)
        nickels = int(((dollars % 0.25) % 0.10) / 0.05)
        pennies = int(((dollars % 0.25) % 0.10) % 0.05)

        coins = coins + quarters
        if dime > 0:
            coins = coins + dime
        if nickels > 0:
            coins = coins + nickels
        if pennies > 0:
            coins = coins + pennies

        print(coins)
        break

File: cash/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    float change;
    int coins = 0;

    do
    {
        change = get_float("change owed: ");
    }

    while (change < 0);

    while (change >= 25)
    {
        coins++;
        change = change - 25;
    }

    while (change >= 10)
    {
        coins++;
        change = change - 10;
    }

    while (change >= 5)
    {
        coins++;
        change = change - 5;
    }

    while (change >= 1)
    {
        coins++;
        change = change - 1;
    }

    printf("%d\n", coins);
}
